torch/_subclasses/meta_utils.py

```diff
@@ -1669,6 +1669,8 @@ def is_c_of_r(
                             torch._C.DispatchKey.ADInplaceOrView, old_exclude
                         )
 
+                    r.fake_device = t.device  # type: ignore[attr-defined]
```

Comment:
It is interesting that this has never been triggered before. For the base and view to have differing devices is truly cursed...